### PR TITLE
Move legacy request helpers

### PR DESCRIPTION
**Pull request by paulmelnikow**
In #2698 we decided to put legacy helper functions in `core/legacy`. I think that’s a fine idea, though if we’re going to have a bunch of badge helper functions in there, it seems like it is probably better to keep these two important but esoteric helper functions with the core code to which they are most coupled. So I added `legacy-` to the name, and put them in `core/base-service`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -89,7 +89,6 @@ There are three places to get help:
   used by developers or which are widely used by developers
 - The left-hand side of a badge should not advertise. It should be a lowercase _noun_
   succinctly describing the meaning of the right-hand side.
-- Query parameters must be _declared by the service_. See `request-handler.js`.
```
Comment (paulmelnikow) [inline CONTRIBUTING.md]: This note predates `BaseService` and I think is obsoleted by our (albeit fledgling) docs in BaseService.

File: core/base-service/legacy-request-handler.js [hunk 2]
```diff
@@ -4,18 +4,18 @@
...
+const makeBadge = require('../../gh-badges/lib/make-badge')
 const {
   Inaccessible,
   InvalidResponse,
   ShieldsRuntimeError,
-} = require('../services/errors')
-const { setCacheHeaders } = require('../services/cache-headers')
-const { makeBadgeData: getBadgeData } = require('./badge-data')
-const { makeSend } = require('./result-sender')
+} = require('../../services/errors')
```
[the remark this answers is not in the document]
Comment (paulmelnikow) [reply core/base-service/legacy-request-handler.js]: I want to be careful to avoid circular imports, though since this is only used from `core/server/server.js` I think that would be fine.

[the remark this answers is not in the document]
Comment (calebcartwright) [reply core/base-service/legacy-request-handler.js]: ah! yeah probably better safe than sorry. I'm still hoping that the legacy stuff will all be refactored soon

[the remark this answers is not in the document]
Comment (paulmelnikow) [reply core/base-service/legacy-request-handler.js]: I hope so! I mean, I think a new version of this functionality (i.e. caching and timeout) could be written for the modern services. I don't think it will be worth it, though, to refactor the legacy services not to use this. So we'll have those to deal with, anyway. We still have 75 files with legacy services…

I was talking with @jankeromnes about doing a distributed hack day to implement new badges, though maybe it would make more sense to get people to try to duck in and refactor. I feel like once someone has done one or two of these, they can burn through them pretty effectively.

[the remark this answers is not in the document]
Comment (paulmelnikow) [reply core/base-service/legacy-request-handler.js]: The method in BaseService which adds the route to camp is pretty gnarly. You'll probably recognize its structure from the legacy services you've worked on:

https://github.com/badges/shields/blob/aac2a4d5ad5f1a1494e1082a6da2a70e358643c0/services/base.js#L490-L518

It's really hard to follow. There's a call to `handleRequest` near the top which is defined in `legacy-request-handler`. Code in there delegates back to the `handler` callback that's defined here. That invokes the service, which will eventually pass control back to the caching request via `await this._request()`. (That invokes `request.isPromise()` from the `request` variable being injected here.) When the service is done it eventually invokes `sendBadge`, passing control back to code in `legacy-request-handler`.
